jina/drivers/__init__.py

```diff
@@ -329,10 +329,14 @@ def pre_traverse(_docs, recur_on, context_doc=None,
             for d in docs:
                 # Move to starting depth range
                 # assume that these search docs have a maximum of one chunk per document (common pattern in search)
-                working_doc = d
-                while working_doc.granularity < self._granularity_start:
-                    working_doc = working_doc.chunks[0]
-                _traverse(working_doc.matches, 'matches', None, 'adjacency', self._adjacency_start, self._adjacency_end)
+                working_docs = [d]
+                while working_docs[0].granularity < self._granularity_start:
```

Comment:
Smart move

Reply:
If we go for this, please remember to adapt the tests I added to have more chunks per query doc.